// @aws-sdk/client-cloudformation/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LimitExceededException = exports.InsufficientCapabilitiesException = exports.CreateChangeSetOutput = exports.CreateChangeSetInput = exports.Tag = exports.RollbackConfiguration = exports.RollbackTrigger = exports.ResourceToImport = exports.Parameter = exports.ContinueUpdateRollbackOutput = exports.ContinueUpdateRollbackInput = exports.ChangeSetType = exports.ChangeSetSummary = exports.ExecutionStatus = exports.ChangeSetStatus = exports.ChangeSetNotFoundException = exports.Change = exports.ChangeType = exports.ResourceChange = exports.Replacement = exports.ModuleInfo = exports.ResourceChangeDetail = exports.ResourceTargetDefinition = exports.RequiresRecreation = exports.ResourceAttribute = exports.EvaluationType = exports.ChangeSource = exports.ChangeAction = exports.Category = exports.Capability = exports.TokenAlreadyExistsException = exports.CancelUpdateStackInput = exports.CallAs = exports.TypeConfigurationNotFoundException = exports.BatchDescribeTypeConfigurationsOutput = exports.TypeConfigurationDetails = exports.BatchDescribeTypeConfigurationsError = exports.BatchDescribeTypeConfigurationsInput = exports.TypeConfigurationIdentifier = exports.AutoDeployment = exports.AlreadyExistsException = exports.TypeNotFoundException = exports.CFNRegistryException = exports.ActivateTypeOutput = exports.ActivateTypeInput = exports.VersionBump = exports.ThirdPartyType = exports.LoggingConfig = exports.AccountLimit = exports.AccountGateResult = void 0;
exports.DescribeStackInstanceOutput = exports.StackInstance = exports.StackInstanceComprehensiveStatus = exports.DescribeStackInstanceInput = exports.DescribeStackEventsOutput = exports.StackEvent = exports.ResourceStatus = exports.DescribeStackEventsInput = exports.DescribeStackDriftDetectionStatusOutput = exports.StackDriftStatus = exports.StackDriftDetectionStatus = exports.DescribeStackDriftDetectionStatusInput = exports.DescribePublisherOutput = exports.PublisherStatus = exports.IdentityProvider = exports.DescribePublisherInput = exports.DescribeChangeSetOutput = exports.DescribeChangeSetInput = exports.DescribeAccountLimitsOutput = exports.DescribeAccountLimitsInput = exports.DeregisterTypeOutput = exports.DeregisterTypeInput = exports.StackSetNotEmptyException = exports.DeleteStackSetOutput = exports.DeleteStackSetInput = exports.DeleteStackInstancesOutput = exports.DeleteStackInstancesInput = exports.DeleteStackInput = exports.InvalidChangeSetStatusException = exports.DeleteChangeSetOutput = exports.DeleteChangeSetInput = exports.DeactivateTypeOutput = exports.DeactivateTypeInput = exports.NameAlreadyExistsException = exports.CreateStackSetOutput = exports.CreateStackSetInput = exports.CreatedButModifiedException = exports.StaleRequestException = exports.StackSetNotFoundException = exports.OperationInProgressException = exports.OperationIdAlreadyExistsException = exports.InvalidOperationException = exports.CreateStackInstancesOutput = exports.CreateStackInstancesInput = exports.StackSetOperationPreferences = exports.RegionConcurrencyType = exports.DeploymentTargets = exports.CreateStackOutput = exports.CreateStackInput = exports.OnFailure = void 0;
exports.TemplateStage = exports.GetStackPolicyOutput = exports.GetStackPolicyInput = exports.ExecuteChangeSetOutput = exports.ExecuteChangeSetInput = exports.EstimateTemplateCostOutput = exports.EstimateTemplateCostInput = exports.DetectStackSetDriftOutput = exports.DetectStackSetDriftInput = exports.DetectStackResourceDriftOutput = exports.DetectStackResourceDriftInput = exports.DetectStackDriftOutput = exports.DetectStackDriftInput = exports.DescribeTypeRegistrationOutput = exports.DescribeTypeRegistrationInput = exports.DescribeTypeOutput = exports.TypeTestsStatus = exports.RequiredActivatedType = exports.DescribeTypeInput = exports.OperationNotFoundException = exports.DescribeStackSetOperationOutput = exports.StackSetOperation = exports.DescribeStackSetOperationInput = exports.DescribeStackSetOutput = exports.StackSet = exports.StackSetDriftDetectionDetails = exports.StackSetDriftStatus = exports.StackSetDriftDetectionStatus = exports.DescribeStackSetInput = exports.DescribeStacksOutput = exports.Stack = exports.StackStatus = exports.Output = exports.StackDriftInformation = exports.DescribeStacksInput = exports.DescribeStackResourcesOutput = exports.StackResource = exports.DescribeStackResourcesInput = exports.DescribeStackResourceDriftsOutput = exports.StackResourceDrift = exports.PropertyDifference = exports.DifferenceType = exports.PhysicalResourceIdContextKeyValuePair = exports.DescribeStackResourceDriftsInput = exports.DescribeStackResourceOutput = exports.StackResourceDetail = exports.StackResourceDriftInformation = exports.StackResourceDriftStatus = exports.DescribeStackResourceInput = exports.StackInstanceNotFoundException = void 0;
exports.InvalidStateTransitionException = exports.PublishTypeOutput = exports.PublishTypeInput = exports.ListTypeVersionsOutput = exports.TypeVersionSummary = exports.ListTypeVersionsInput = exports.ListTypesOutput = exports.TypeSummary = exports.ListTypesInput = exports.TypeFilters = exports.ListTypeRegistrationsOutput = exports.ListTypeRegistrationsInput = exports.ListStackSetsOutput = exports.StackSetSummary = exports.ListStackSetsInput = exports.ListStackSetOperationsOutput = exports.StackSetOperationSummary = exports.ListStackSetOperationsInput = exports.ListStackSetOperationResultsOutput = exports.StackSetOperationResultSummary = exports.ListStackSetOperationResultsInput = exports.ListStacksOutput = exports.StackSummary = exports.StackDriftInformationSummary = exports.ListStacksInput = exports.ListStackResourcesOutput = exports.StackResourceSummary = exports.StackResourceDriftInformationSummary = exports.ListStackResourcesInput = exports.ListStackInstancesOutput = exports.StackInstanceSummary = exports.ListStackInstancesInput = exports.StackInstanceFilter = exports.ListImportsOutput = exports.ListImportsInput = exports.ListExportsOutput = exports.Export = exports.ListExportsInput = exports.ListChangeSetsOutput = exports.ListChangeSetsInput = exports.StackNotFoundException = exports.ImportStacksToStackSetOutput = exports.ImportStacksToStackSetInput = exports.GetTemplateSummaryOutput = exports.ResourceIdentifierSummary = exports.ParameterDeclaration = exports.ParameterConstraints = exports.GetTemplateSummaryInput = exports.GetTemplateOutput = exports.GetTemplateInput = void 0;
exports.ValidateTemplateOutput = exports.TemplateParameter = exports.ValidateTemplateInput = exports.UpdateTerminationProtectionOutput = exports.UpdateTerminationProtectionInput = exports.UpdateStackSetOutput = exports.UpdateStackSetInput = exports.UpdateStackInstancesOutput = exports.UpdateStackInstancesInput = exports.UpdateStackOutput = exports.UpdateStackInput = exports.TestTypeOutput = exports.TestTypeInput = exports.StopStackSetOperationOutput = exports.StopStackSetOperationInput = exports.SignalResourceInput = exports.ResourceSignalStatus = exports.SetTypeDefaultVersionOutput = exports.SetTypeDefaultVersionInput = exports.SetTypeConfigurationOutput = exports.SetTypeConfigurationInput = exports.SetStackPolicyInput = exports.RollbackStackOutput = exports.RollbackStackInput = exports.RegisterTypeOutput = exports.RegisterTypeInput = exports.RegisterPublisherOutput = exports.RegisterPublisherInput = exports.RecordHandlerProgressOutput = exports.RecordHandlerProgressInput = exports.HandlerErrorCode = exports.OperationStatus = exports.OperationStatusCheckFailedException = void 0;
var AccountGateResult;
(function (AccountGateResult) {
    AccountGateResult.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccountGateResult = exports.AccountGateResult || (exports.AccountGateResult = {}));
var AccountLimit;
(function (AccountLimit) {
    AccountLimit.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AccountLimit = exports.AccountLimit || (exports.AccountLimit = {}));
var LoggingConfig;
(function (LoggingConfig) {
    LoggingConfig.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LoggingConfig = exports.LoggingConfig || (exports.LoggingConfig = {}));
var ThirdPartyType;
(function (ThirdPartyType) {
    ThirdPartyType["MODULE"] = "MODULE";
    ThirdPartyType["RESOURCE"] = "RESOURCE";
})(ThirdPartyType = exports.ThirdPartyType || (exports.ThirdPartyType = {}));
var VersionBump;
(function (VersionBump) {
    VersionBump["MAJOR"] = "MAJOR";
    VersionBump["MINOR"] = "MINOR";
})(VersionBump = exports.VersionBump || (exports.VersionBump = {}));
var ActivateTypeInput;
(function (ActivateTypeInput) {
    ActivateTypeInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ActivateTypeInput = exports.ActivateTypeInput || (exports.ActivateTypeInput = {}));
var ActivateTypeOutput;
(function (ActivateTypeOutput) {
    ActivateTypeOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ActivateTypeOutput = exports.ActivateTypeOutput || (exports.ActivateTypeOutput = {}));
var CFNRegistryException;
(function (CFNRegistryException) {
    CFNRegistryException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CFNRegistryException = exports.CFNRegistryException || (exports.CFNRegistryException = {}));
var TypeNotFoundException;
(function (TypeNotFoundException) {
    TypeNotFoundException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TypeNotFoundException = exports.TypeNotFoundException || (exports.TypeNotFoundException = {}));
var AlreadyExistsException;
(function (AlreadyExistsException) {
    AlreadyExistsException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AlreadyExistsException = exports.AlreadyExistsException || (exports.AlreadyExistsException = {}));
var AutoDeployment;
(function (AutoDeployment) {
    AutoDeployment.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(AutoDeployment = exports.AutoDeployment || (exports.AutoDeployment = {}));
var TypeConfigurationIdentifier;
(function (TypeConfigurationIdentifier) {
    TypeConfigurationIdentifier.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TypeConfigurationIdentifier = exports.TypeConfigurationIdentifier || (exports.TypeConfigurationIdentifier = {}));
var BatchDescribeTypeConfigurationsInput;
(function (BatchDescribeTypeConfigurationsInput) {
    BatchDescribeTypeConfigurationsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BatchDescribeTypeConfigurationsInput = exports.BatchDescribeTypeConfigurationsInput || (exports.BatchDescribeTypeConfigurationsInput = {}));
var BatchDescribeTypeConfigurationsError;
(function (BatchDescribeTypeConfigurationsError) {
    BatchDescribeTypeConfigurationsError.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BatchDescribeTypeConfigurationsError = exports.BatchDescribeTypeConfigurationsError || (exports.BatchDescribeTypeConfigurationsError = {}));
var TypeConfigurationDetails;
(function (TypeConfigurationDetails) {
    TypeConfigurationDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TypeConfigurationDetails = exports.TypeConfigurationDetails || (exports.TypeConfigurationDetails = {}));
var BatchDescribeTypeConfigurationsOutput;
(function (BatchDescribeTypeConfigurationsOutput) {
    BatchDescribeTypeConfigurationsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(BatchDescribeTypeConfigurationsOutput = exports.BatchDescribeTypeConfigurationsOutput || (exports.BatchDescribeTypeConfigurationsOutput = {}));
var TypeConfigurationNotFoundException;
(function (TypeConfigurationNotFoundException) {
    TypeConfigurationNotFoundException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TypeConfigurationNotFoundException = exports.TypeConfigurationNotFoundException || (exports.TypeConfigurationNotFoundException = {}));
var CallAs;
(function (CallAs) {
    CallAs["DELEGATED_ADMIN"] = "DELEGATED_ADMIN";
    CallAs["SELF"] = "SELF";
})(CallAs = exports.CallAs || (exports.CallAs = {}));
var CancelUpdateStackInput;
(function (CancelUpdateStackInput) {
    CancelUpdateStackInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CancelUpdateStackInput = exports.CancelUpdateStackInput || (exports.CancelUpdateStackInput = {}));
var TokenAlreadyExistsException;
(function (TokenAlreadyExistsException) {
    TokenAlreadyExistsException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TokenAlreadyExistsException = exports.TokenAlreadyExistsException || (exports.TokenAlreadyExistsException = {}));
var Capability;
(function (Capability) {
    Capability["CAPABILITY_AUTO_EXPAND"] = "CAPABILITY_AUTO_EXPAND";
    Capability["CAPABILITY_IAM"] = "CAPABILITY_IAM";
    Capability["CAPABILITY_NAMED_IAM"] = "CAPABILITY_NAMED_IAM";
})(Capability = exports.Capability || (exports.Capability = {}));
var Category;
(function (Category) {
    Category["ACTIVATED"] = "ACTIVATED";
    Category["AWS_TYPES"] = "AWS_TYPES";
    Category["REGISTERED"] = "REGISTERED";
    Category["THIRD_PARTY"] = "THIRD_PARTY";
})(Category = exports.Category || (exports.Category = {}));
var ChangeAction;
(function (ChangeAction) {
    ChangeAction["Add"] = "Add";
    ChangeAction["Dynamic"] = "Dynamic";
    ChangeAction["Import"] = "Import";
    ChangeAction["Modify"] = "Modify";
    ChangeAction["Remove"] = "Remove";
})(ChangeAction = exports.ChangeAction || (exports.ChangeAction = {}));
var ChangeSource;
(function (ChangeSource) {
    ChangeSource["Automatic"] = "Automatic";
    ChangeSource["DirectModification"] = "DirectModification";
    ChangeSource["ParameterReference"] = "ParameterReference";
    ChangeSource["ResourceAttribute"] = "ResourceAttribute";
    ChangeSource["ResourceReference"] = "ResourceReference";
})(ChangeSource = exports.ChangeSource || (exports.ChangeSource = {}));
var EvaluationType;
(function (EvaluationType) {
    EvaluationType["Dynamic"] = "Dynamic";
    EvaluationType["Static"] = "Static";
})(EvaluationType = exports.EvaluationType || (exports.EvaluationType = {}));
var ResourceAttribute;
(function (ResourceAttribute) {
    ResourceAttribute["CreationPolicy"] = "CreationPolicy";
    ResourceAttribute["DeletionPolicy"] = "DeletionPolicy";
    ResourceAttribute["Metadata"] = "Metadata";
    ResourceAttribute["Properties"] = "Properties";
    ResourceAttribute["Tags"] = "Tags";
    ResourceAttribute["UpdatePolicy"] = "UpdatePolicy";
})(ResourceAttribute = exports.ResourceAttribute || (exports.ResourceAttribute = {}));
var RequiresRecreation;
(function (RequiresRecreation) {
    RequiresRecreation["Always"] = "Always";
    RequiresRecreation["Conditionally"] = "Conditionally";
    RequiresRecreation["Never"] = "Never";
})(RequiresRecreation = exports.RequiresRecreation || (exports.RequiresRecreation = {}));
var ResourceTargetDefinition;
(function (ResourceTargetDefinition) {
    ResourceTargetDefinition.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceTargetDefinition = exports.ResourceTargetDefinition || (exports.ResourceTargetDefinition = {}));
var ResourceChangeDetail;
(function (ResourceChangeDetail) {
    ResourceChangeDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceChangeDetail = exports.ResourceChangeDetail || (exports.ResourceChangeDetail = {}));
var ModuleInfo;
(function (ModuleInfo) {
    ModuleInfo.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ModuleInfo = exports.ModuleInfo || (exports.ModuleInfo = {}));
var Replacement;
(function (Replacement) {
    Replacement["Conditional"] = "Conditional";
    Replacement["False"] = "False";
    Replacement["True"] = "True";
})(Replacement = exports.Replacement || (exports.Replacement = {}));
var ResourceChange;
(function (ResourceChange) {
    ResourceChange.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceChange = exports.ResourceChange || (exports.ResourceChange = {}));
var ChangeType;
(function (ChangeType) {
    ChangeType["Resource"] = "Resource";
})(ChangeType = exports.ChangeType || (exports.ChangeType = {}));
var Change;
(function (Change) {
    Change.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Change = exports.Change || (exports.Change = {}));
var ChangeSetNotFoundException;
(function (ChangeSetNotFoundException) {
    ChangeSetNotFoundException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeSetNotFoundException = exports.ChangeSetNotFoundException || (exports.ChangeSetNotFoundException = {}));
var ChangeSetStatus;
(function (ChangeSetStatus) {
    ChangeSetStatus["CREATE_COMPLETE"] = "CREATE_COMPLETE";
    ChangeSetStatus["CREATE_IN_PROGRESS"] = "CREATE_IN_PROGRESS";
    ChangeSetStatus["CREATE_PENDING"] = "CREATE_PENDING";
    ChangeSetStatus["DELETE_COMPLETE"] = "DELETE_COMPLETE";
    ChangeSetStatus["DELETE_FAILED"] = "DELETE_FAILED";
    ChangeSetStatus["DELETE_IN_PROGRESS"] = "DELETE_IN_PROGRESS";
    ChangeSetStatus["DELETE_PENDING"] = "DELETE_PENDING";
    ChangeSetStatus["FAILED"] = "FAILED";
})(ChangeSetStatus = exports.ChangeSetStatus || (exports.ChangeSetStatus = {}));
var ExecutionStatus;
(function (ExecutionStatus) {
    ExecutionStatus["AVAILABLE"] = "AVAILABLE";
    ExecutionStatus["EXECUTE_COMPLETE"] = "EXECUTE_COMPLETE";
    ExecutionStatus["EXECUTE_FAILED"] = "EXECUTE_FAILED";
    ExecutionStatus["EXECUTE_IN_PROGRESS"] = "EXECUTE_IN_PROGRESS";
    ExecutionStatus["OBSOLETE"] = "OBSOLETE";
    ExecutionStatus["UNAVAILABLE"] = "UNAVAILABLE";
})(ExecutionStatus = exports.ExecutionStatus || (exports.ExecutionStatus = {}));
var ChangeSetSummary;
(function (ChangeSetSummary) {
    ChangeSetSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ChangeSetSummary = exports.ChangeSetSummary || (exports.ChangeSetSummary = {}));
var ChangeSetType;
(function (ChangeSetType) {
    ChangeSetType["CREATE"] = "CREATE";
    ChangeSetType["IMPORT"] = "IMPORT";
    ChangeSetType["UPDATE"] = "UPDATE";
})(ChangeSetType = exports.ChangeSetType || (exports.ChangeSetType = {}));
var ContinueUpdateRollbackInput;
(function (ContinueUpdateRollbackInput) {
    ContinueUpdateRollbackInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ContinueUpdateRollbackInput = exports.ContinueUpdateRollbackInput || (exports.ContinueUpdateRollbackInput = {}));
var ContinueUpdateRollbackOutput;
(function (ContinueUpdateRollbackOutput) {
    ContinueUpdateRollbackOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ContinueUpdateRollbackOutput = exports.ContinueUpdateRollbackOutput || (exports.ContinueUpdateRollbackOutput = {}));
var Parameter;
(function (Parameter) {
    Parameter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Parameter = exports.Parameter || (exports.Parameter = {}));
var ResourceToImport;
(function (ResourceToImport) {
    ResourceToImport.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceToImport = exports.ResourceToImport || (exports.ResourceToImport = {}));
var RollbackTrigger;
(function (RollbackTrigger) {
    RollbackTrigger.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RollbackTrigger = exports.RollbackTrigger || (exports.RollbackTrigger = {}));
var RollbackConfiguration;
(function (RollbackConfiguration) {
    RollbackConfiguration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RollbackConfiguration = exports.RollbackConfiguration || (exports.RollbackConfiguration = {}));
var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Tag = exports.Tag || (exports.Tag = {}));
var CreateChangeSetInput;
(function (CreateChangeSetInput) {
    CreateChangeSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateChangeSetInput = exports.CreateChangeSetInput || (exports.CreateChangeSetInput = {}));
var CreateChangeSetOutput;
(function (CreateChangeSetOutput) {
    CreateChangeSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateChangeSetOutput = exports.CreateChangeSetOutput || (exports.CreateChangeSetOutput = {}));
var InsufficientCapabilitiesException;
(function (InsufficientCapabilitiesException) {
    InsufficientCapabilitiesException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InsufficientCapabilitiesException = exports.InsufficientCapabilitiesException || (exports.InsufficientCapabilitiesException = {}));
var LimitExceededException;
(function (LimitExceededException) {
    LimitExceededException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(LimitExceededException = exports.LimitExceededException || (exports.LimitExceededException = {}));
var OnFailure;
(function (OnFailure) {
    OnFailure["DELETE"] = "DELETE";
    OnFailure["DO_NOTHING"] = "DO_NOTHING";
    OnFailure["ROLLBACK"] = "ROLLBACK";
})(OnFailure = exports.OnFailure || (exports.OnFailure = {}));
var CreateStackInput;
(function (CreateStackInput) {
    CreateStackInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStackInput = exports.CreateStackInput || (exports.CreateStackInput = {}));
var CreateStackOutput;
(function (CreateStackOutput) {
    CreateStackOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStackOutput = exports.CreateStackOutput || (exports.CreateStackOutput = {}));
var DeploymentTargets;
(function (DeploymentTargets) {
    DeploymentTargets.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeploymentTargets = exports.DeploymentTargets || (exports.DeploymentTargets = {}));
var RegionConcurrencyType;
(function (RegionConcurrencyType) {
    RegionConcurrencyType["PARALLEL"] = "PARALLEL";
    RegionConcurrencyType["SEQUENTIAL"] = "SEQUENTIAL";
})(RegionConcurrencyType = exports.RegionConcurrencyType || (exports.RegionConcurrencyType = {}));
var StackSetOperationPreferences;
(function (StackSetOperationPreferences) {
    StackSetOperationPreferences.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetOperationPreferences = exports.StackSetOperationPreferences || (exports.StackSetOperationPreferences = {}));
var CreateStackInstancesInput;
(function (CreateStackInstancesInput) {
    CreateStackInstancesInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStackInstancesInput = exports.CreateStackInstancesInput || (exports.CreateStackInstancesInput = {}));
var CreateStackInstancesOutput;
(function (CreateStackInstancesOutput) {
    CreateStackInstancesOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStackInstancesOutput = exports.CreateStackInstancesOutput || (exports.CreateStackInstancesOutput = {}));
var InvalidOperationException;
(function (InvalidOperationException) {
    InvalidOperationException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidOperationException = exports.InvalidOperationException || (exports.InvalidOperationException = {}));
var OperationIdAlreadyExistsException;
(function (OperationIdAlreadyExistsException) {
    OperationIdAlreadyExistsException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OperationIdAlreadyExistsException = exports.OperationIdAlreadyExistsException || (exports.OperationIdAlreadyExistsException = {}));
var OperationInProgressException;
(function (OperationInProgressException) {
    OperationInProgressException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OperationInProgressException = exports.OperationInProgressException || (exports.OperationInProgressException = {}));
var StackSetNotFoundException;
(function (StackSetNotFoundException) {
    StackSetNotFoundException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetNotFoundException = exports.StackSetNotFoundException || (exports.StackSetNotFoundException = {}));
var StaleRequestException;
(function (StaleRequestException) {
    StaleRequestException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StaleRequestException = exports.StaleRequestException || (exports.StaleRequestException = {}));
var CreatedButModifiedException;
(function (CreatedButModifiedException) {
    CreatedButModifiedException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreatedButModifiedException = exports.CreatedButModifiedException || (exports.CreatedButModifiedException = {}));
var CreateStackSetInput;
(function (CreateStackSetInput) {
    CreateStackSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStackSetInput = exports.CreateStackSetInput || (exports.CreateStackSetInput = {}));
var CreateStackSetOutput;
(function (CreateStackSetOutput) {
    CreateStackSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(CreateStackSetOutput = exports.CreateStackSetOutput || (exports.CreateStackSetOutput = {}));
var NameAlreadyExistsException;
(function (NameAlreadyExistsException) {
    NameAlreadyExistsException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(NameAlreadyExistsException = exports.NameAlreadyExistsException || (exports.NameAlreadyExistsException = {}));
var DeactivateTypeInput;
(function (DeactivateTypeInput) {
    DeactivateTypeInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeactivateTypeInput = exports.DeactivateTypeInput || (exports.DeactivateTypeInput = {}));
var DeactivateTypeOutput;
(function (DeactivateTypeOutput) {
    DeactivateTypeOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeactivateTypeOutput = exports.DeactivateTypeOutput || (exports.DeactivateTypeOutput = {}));
var DeleteChangeSetInput;
(function (DeleteChangeSetInput) {
    DeleteChangeSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteChangeSetInput = exports.DeleteChangeSetInput || (exports.DeleteChangeSetInput = {}));
var DeleteChangeSetOutput;
(function (DeleteChangeSetOutput) {
    DeleteChangeSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteChangeSetOutput = exports.DeleteChangeSetOutput || (exports.DeleteChangeSetOutput = {}));
var InvalidChangeSetStatusException;
(function (InvalidChangeSetStatusException) {
    InvalidChangeSetStatusException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidChangeSetStatusException = exports.InvalidChangeSetStatusException || (exports.InvalidChangeSetStatusException = {}));
var DeleteStackInput;
(function (DeleteStackInput) {
    DeleteStackInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStackInput = exports.DeleteStackInput || (exports.DeleteStackInput = {}));
var DeleteStackInstancesInput;
(function (DeleteStackInstancesInput) {
    DeleteStackInstancesInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStackInstancesInput = exports.DeleteStackInstancesInput || (exports.DeleteStackInstancesInput = {}));
var DeleteStackInstancesOutput;
(function (DeleteStackInstancesOutput) {
    DeleteStackInstancesOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStackInstancesOutput = exports.DeleteStackInstancesOutput || (exports.DeleteStackInstancesOutput = {}));
var DeleteStackSetInput;
(function (DeleteStackSetInput) {
    DeleteStackSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStackSetInput = exports.DeleteStackSetInput || (exports.DeleteStackSetInput = {}));
var DeleteStackSetOutput;
(function (DeleteStackSetOutput) {
    DeleteStackSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeleteStackSetOutput = exports.DeleteStackSetOutput || (exports.DeleteStackSetOutput = {}));
var StackSetNotEmptyException;
(function (StackSetNotEmptyException) {
    StackSetNotEmptyException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetNotEmptyException = exports.StackSetNotEmptyException || (exports.StackSetNotEmptyException = {}));
var DeregisterTypeInput;
(function (DeregisterTypeInput) {
    DeregisterTypeInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeregisterTypeInput = exports.DeregisterTypeInput || (exports.DeregisterTypeInput = {}));
var DeregisterTypeOutput;
(function (DeregisterTypeOutput) {
    DeregisterTypeOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DeregisterTypeOutput = exports.DeregisterTypeOutput || (exports.DeregisterTypeOutput = {}));
var DescribeAccountLimitsInput;
(function (DescribeAccountLimitsInput) {
    DescribeAccountLimitsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAccountLimitsInput = exports.DescribeAccountLimitsInput || (exports.DescribeAccountLimitsInput = {}));
var DescribeAccountLimitsOutput;
(function (DescribeAccountLimitsOutput) {
    DescribeAccountLimitsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeAccountLimitsOutput = exports.DescribeAccountLimitsOutput || (exports.DescribeAccountLimitsOutput = {}));
var DescribeChangeSetInput;
(function (DescribeChangeSetInput) {
    DescribeChangeSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeChangeSetInput = exports.DescribeChangeSetInput || (exports.DescribeChangeSetInput = {}));
var DescribeChangeSetOutput;
(function (DescribeChangeSetOutput) {
    DescribeChangeSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeChangeSetOutput = exports.DescribeChangeSetOutput || (exports.DescribeChangeSetOutput = {}));
var DescribePublisherInput;
(function (DescribePublisherInput) {
    DescribePublisherInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribePublisherInput = exports.DescribePublisherInput || (exports.DescribePublisherInput = {}));
var IdentityProvider;
(function (IdentityProvider) {
    IdentityProvider["AWS_Marketplace"] = "AWS_Marketplace";
    IdentityProvider["Bitbucket"] = "Bitbucket";
    IdentityProvider["GitHub"] = "GitHub";
})(IdentityProvider = exports.IdentityProvider || (exports.IdentityProvider = {}));
var PublisherStatus;
(function (PublisherStatus) {
    PublisherStatus["UNVERIFIED"] = "UNVERIFIED";
    PublisherStatus["VERIFIED"] = "VERIFIED";
})(PublisherStatus = exports.PublisherStatus || (exports.PublisherStatus = {}));
var DescribePublisherOutput;
(function (DescribePublisherOutput) {
    DescribePublisherOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribePublisherOutput = exports.DescribePublisherOutput || (exports.DescribePublisherOutput = {}));
var DescribeStackDriftDetectionStatusInput;
(function (DescribeStackDriftDetectionStatusInput) {
    DescribeStackDriftDetectionStatusInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackDriftDetectionStatusInput = exports.DescribeStackDriftDetectionStatusInput || (exports.DescribeStackDriftDetectionStatusInput = {}));
var StackDriftDetectionStatus;
(function (StackDriftDetectionStatus) {
    StackDriftDetectionStatus["DETECTION_COMPLETE"] = "DETECTION_COMPLETE";
    StackDriftDetectionStatus["DETECTION_FAILED"] = "DETECTION_FAILED";
    StackDriftDetectionStatus["DETECTION_IN_PROGRESS"] = "DETECTION_IN_PROGRESS";
})(StackDriftDetectionStatus = exports.StackDriftDetectionStatus || (exports.StackDriftDetectionStatus = {}));
var StackDriftStatus;
(function (StackDriftStatus) {
    StackDriftStatus["DRIFTED"] = "DRIFTED";
    StackDriftStatus["IN_SYNC"] = "IN_SYNC";
    StackDriftStatus["NOT_CHECKED"] = "NOT_CHECKED";
    StackDriftStatus["UNKNOWN"] = "UNKNOWN";
})(StackDriftStatus = exports.StackDriftStatus || (exports.StackDriftStatus = {}));
var DescribeStackDriftDetectionStatusOutput;
(function (DescribeStackDriftDetectionStatusOutput) {
    DescribeStackDriftDetectionStatusOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackDriftDetectionStatusOutput = exports.DescribeStackDriftDetectionStatusOutput || (exports.DescribeStackDriftDetectionStatusOutput = {}));
var DescribeStackEventsInput;
(function (DescribeStackEventsInput) {
    DescribeStackEventsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackEventsInput = exports.DescribeStackEventsInput || (exports.DescribeStackEventsInput = {}));
var ResourceStatus;
(function (ResourceStatus) {
    ResourceStatus["CREATE_COMPLETE"] = "CREATE_COMPLETE";
    ResourceStatus["CREATE_FAILED"] = "CREATE_FAILED";
    ResourceStatus["CREATE_IN_PROGRESS"] = "CREATE_IN_PROGRESS";
    ResourceStatus["DELETE_COMPLETE"] = "DELETE_COMPLETE";
    ResourceStatus["DELETE_FAILED"] = "DELETE_FAILED";
    ResourceStatus["DELETE_IN_PROGRESS"] = "DELETE_IN_PROGRESS";
    ResourceStatus["DELETE_SKIPPED"] = "DELETE_SKIPPED";
    ResourceStatus["IMPORT_COMPLETE"] = "IMPORT_COMPLETE";
    ResourceStatus["IMPORT_FAILED"] = "IMPORT_FAILED";
    ResourceStatus["IMPORT_IN_PROGRESS"] = "IMPORT_IN_PROGRESS";
    ResourceStatus["IMPORT_ROLLBACK_COMPLETE"] = "IMPORT_ROLLBACK_COMPLETE";
    ResourceStatus["IMPORT_ROLLBACK_FAILED"] = "IMPORT_ROLLBACK_FAILED";
    ResourceStatus["IMPORT_ROLLBACK_IN_PROGRESS"] = "IMPORT_ROLLBACK_IN_PROGRESS";
    ResourceStatus["ROLLBACK_COMPLETE"] = "ROLLBACK_COMPLETE";
    ResourceStatus["ROLLBACK_FAILED"] = "ROLLBACK_FAILED";
    ResourceStatus["ROLLBACK_IN_PROGRESS"] = "ROLLBACK_IN_PROGRESS";
    ResourceStatus["UPDATE_COMPLETE"] = "UPDATE_COMPLETE";
    ResourceStatus["UPDATE_FAILED"] = "UPDATE_FAILED";
    ResourceStatus["UPDATE_IN_PROGRESS"] = "UPDATE_IN_PROGRESS";
    ResourceStatus["UPDATE_ROLLBACK_COMPLETE"] = "UPDATE_ROLLBACK_COMPLETE";
    ResourceStatus["UPDATE_ROLLBACK_FAILED"] = "UPDATE_ROLLBACK_FAILED";
    ResourceStatus["UPDATE_ROLLBACK_IN_PROGRESS"] = "UPDATE_ROLLBACK_IN_PROGRESS";
})(ResourceStatus = exports.ResourceStatus || (exports.ResourceStatus = {}));
var StackEvent;
(function (StackEvent) {
    StackEvent.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackEvent = exports.StackEvent || (exports.StackEvent = {}));
var DescribeStackEventsOutput;
(function (DescribeStackEventsOutput) {
    DescribeStackEventsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackEventsOutput = exports.DescribeStackEventsOutput || (exports.DescribeStackEventsOutput = {}));
var DescribeStackInstanceInput;
(function (DescribeStackInstanceInput) {
    DescribeStackInstanceInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackInstanceInput = exports.DescribeStackInstanceInput || (exports.DescribeStackInstanceInput = {}));
var StackInstanceComprehensiveStatus;
(function (StackInstanceComprehensiveStatus) {
    StackInstanceComprehensiveStatus.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackInstanceComprehensiveStatus = exports.StackInstanceComprehensiveStatus || (exports.StackInstanceComprehensiveStatus = {}));
var StackInstance;
(function (StackInstance) {
    StackInstance.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackInstance = exports.StackInstance || (exports.StackInstance = {}));
var DescribeStackInstanceOutput;
(function (DescribeStackInstanceOutput) {
    DescribeStackInstanceOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackInstanceOutput = exports.DescribeStackInstanceOutput || (exports.DescribeStackInstanceOutput = {}));
var StackInstanceNotFoundException;
(function (StackInstanceNotFoundException) {
    StackInstanceNotFoundException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackInstanceNotFoundException = exports.StackInstanceNotFoundException || (exports.StackInstanceNotFoundException = {}));
var DescribeStackResourceInput;
(function (DescribeStackResourceInput) {
    DescribeStackResourceInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackResourceInput = exports.DescribeStackResourceInput || (exports.DescribeStackResourceInput = {}));
var StackResourceDriftStatus;
(function (StackResourceDriftStatus) {
    StackResourceDriftStatus["DELETED"] = "DELETED";
    StackResourceDriftStatus["IN_SYNC"] = "IN_SYNC";
    StackResourceDriftStatus["MODIFIED"] = "MODIFIED";
    StackResourceDriftStatus["NOT_CHECKED"] = "NOT_CHECKED";
})(StackResourceDriftStatus = exports.StackResourceDriftStatus || (exports.StackResourceDriftStatus = {}));
var StackResourceDriftInformation;
(function (StackResourceDriftInformation) {
    StackResourceDriftInformation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackResourceDriftInformation = exports.StackResourceDriftInformation || (exports.StackResourceDriftInformation = {}));
var StackResourceDetail;
(function (StackResourceDetail) {
    StackResourceDetail.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackResourceDetail = exports.StackResourceDetail || (exports.StackResourceDetail = {}));
var DescribeStackResourceOutput;
(function (DescribeStackResourceOutput) {
    DescribeStackResourceOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackResourceOutput = exports.DescribeStackResourceOutput || (exports.DescribeStackResourceOutput = {}));
var DescribeStackResourceDriftsInput;
(function (DescribeStackResourceDriftsInput) {
    DescribeStackResourceDriftsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackResourceDriftsInput = exports.DescribeStackResourceDriftsInput || (exports.DescribeStackResourceDriftsInput = {}));
var PhysicalResourceIdContextKeyValuePair;
(function (PhysicalResourceIdContextKeyValuePair) {
    PhysicalResourceIdContextKeyValuePair.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PhysicalResourceIdContextKeyValuePair = exports.PhysicalResourceIdContextKeyValuePair || (exports.PhysicalResourceIdContextKeyValuePair = {}));
var DifferenceType;
(function (DifferenceType) {
    DifferenceType["ADD"] = "ADD";
    DifferenceType["NOT_EQUAL"] = "NOT_EQUAL";
    DifferenceType["REMOVE"] = "REMOVE";
})(DifferenceType = exports.DifferenceType || (exports.DifferenceType = {}));
var PropertyDifference;
(function (PropertyDifference) {
    PropertyDifference.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PropertyDifference = exports.PropertyDifference || (exports.PropertyDifference = {}));
var StackResourceDrift;
(function (StackResourceDrift) {
    StackResourceDrift.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackResourceDrift = exports.StackResourceDrift || (exports.StackResourceDrift = {}));
var DescribeStackResourceDriftsOutput;
(function (DescribeStackResourceDriftsOutput) {
    DescribeStackResourceDriftsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackResourceDriftsOutput = exports.DescribeStackResourceDriftsOutput || (exports.DescribeStackResourceDriftsOutput = {}));
var DescribeStackResourcesInput;
(function (DescribeStackResourcesInput) {
    DescribeStackResourcesInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackResourcesInput = exports.DescribeStackResourcesInput || (exports.DescribeStackResourcesInput = {}));
var StackResource;
(function (StackResource) {
    StackResource.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackResource = exports.StackResource || (exports.StackResource = {}));
var DescribeStackResourcesOutput;
(function (DescribeStackResourcesOutput) {
    DescribeStackResourcesOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackResourcesOutput = exports.DescribeStackResourcesOutput || (exports.DescribeStackResourcesOutput = {}));
var DescribeStacksInput;
(function (DescribeStacksInput) {
    DescribeStacksInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStacksInput = exports.DescribeStacksInput || (exports.DescribeStacksInput = {}));
var StackDriftInformation;
(function (StackDriftInformation) {
    StackDriftInformation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackDriftInformation = exports.StackDriftInformation || (exports.StackDriftInformation = {}));
var Output;
(function (Output) {
    Output.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Output = exports.Output || (exports.Output = {}));
var StackStatus;
(function (StackStatus) {
    StackStatus["CREATE_COMPLETE"] = "CREATE_COMPLETE";
    StackStatus["CREATE_FAILED"] = "CREATE_FAILED";
    StackStatus["CREATE_IN_PROGRESS"] = "CREATE_IN_PROGRESS";
    StackStatus["DELETE_COMPLETE"] = "DELETE_COMPLETE";
    StackStatus["DELETE_FAILED"] = "DELETE_FAILED";
    StackStatus["DELETE_IN_PROGRESS"] = "DELETE_IN_PROGRESS";
    StackStatus["IMPORT_COMPLETE"] = "IMPORT_COMPLETE";
    StackStatus["IMPORT_IN_PROGRESS"] = "IMPORT_IN_PROGRESS";
    StackStatus["IMPORT_ROLLBACK_COMPLETE"] = "IMPORT_ROLLBACK_COMPLETE";
    StackStatus["IMPORT_ROLLBACK_FAILED"] = "IMPORT_ROLLBACK_FAILED";
    StackStatus["IMPORT_ROLLBACK_IN_PROGRESS"] = "IMPORT_ROLLBACK_IN_PROGRESS";
    StackStatus["REVIEW_IN_PROGRESS"] = "REVIEW_IN_PROGRESS";
    StackStatus["ROLLBACK_COMPLETE"] = "ROLLBACK_COMPLETE";
    StackStatus["ROLLBACK_FAILED"] = "ROLLBACK_FAILED";
    StackStatus["ROLLBACK_IN_PROGRESS"] = "ROLLBACK_IN_PROGRESS";
    StackStatus["UPDATE_COMPLETE"] = "UPDATE_COMPLETE";
    StackStatus["UPDATE_COMPLETE_CLEANUP_IN_PROGRESS"] = "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS";
    StackStatus["UPDATE_FAILED"] = "UPDATE_FAILED";
    StackStatus["UPDATE_IN_PROGRESS"] = "UPDATE_IN_PROGRESS";
    StackStatus["UPDATE_ROLLBACK_COMPLETE"] = "UPDATE_ROLLBACK_COMPLETE";
    StackStatus["UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS"] = "UPDATE_ROLLBACK_COMPLETE_CLEANUP_IN_PROGRESS";
    StackStatus["UPDATE_ROLLBACK_FAILED"] = "UPDATE_ROLLBACK_FAILED";
    StackStatus["UPDATE_ROLLBACK_IN_PROGRESS"] = "UPDATE_ROLLBACK_IN_PROGRESS";
})(StackStatus = exports.StackStatus || (exports.StackStatus = {}));
var Stack;
(function (Stack) {
    Stack.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Stack = exports.Stack || (exports.Stack = {}));
var DescribeStacksOutput;
(function (DescribeStacksOutput) {
    DescribeStacksOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStacksOutput = exports.DescribeStacksOutput || (exports.DescribeStacksOutput = {}));
var DescribeStackSetInput;
(function (DescribeStackSetInput) {
    DescribeStackSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackSetInput = exports.DescribeStackSetInput || (exports.DescribeStackSetInput = {}));
var StackSetDriftDetectionStatus;
(function (StackSetDriftDetectionStatus) {
    StackSetDriftDetectionStatus["COMPLETED"] = "COMPLETED";
    StackSetDriftDetectionStatus["FAILED"] = "FAILED";
    StackSetDriftDetectionStatus["IN_PROGRESS"] = "IN_PROGRESS";
    StackSetDriftDetectionStatus["PARTIAL_SUCCESS"] = "PARTIAL_SUCCESS";
    StackSetDriftDetectionStatus["STOPPED"] = "STOPPED";
})(StackSetDriftDetectionStatus = exports.StackSetDriftDetectionStatus || (exports.StackSetDriftDetectionStatus = {}));
var StackSetDriftStatus;
(function (StackSetDriftStatus) {
    StackSetDriftStatus["DRIFTED"] = "DRIFTED";
    StackSetDriftStatus["IN_SYNC"] = "IN_SYNC";
    StackSetDriftStatus["NOT_CHECKED"] = "NOT_CHECKED";
})(StackSetDriftStatus = exports.StackSetDriftStatus || (exports.StackSetDriftStatus = {}));
var StackSetDriftDetectionDetails;
(function (StackSetDriftDetectionDetails) {
    StackSetDriftDetectionDetails.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetDriftDetectionDetails = exports.StackSetDriftDetectionDetails || (exports.StackSetDriftDetectionDetails = {}));
var StackSet;
(function (StackSet) {
    StackSet.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSet = exports.StackSet || (exports.StackSet = {}));
var DescribeStackSetOutput;
(function (DescribeStackSetOutput) {
    DescribeStackSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackSetOutput = exports.DescribeStackSetOutput || (exports.DescribeStackSetOutput = {}));
var DescribeStackSetOperationInput;
(function (DescribeStackSetOperationInput) {
    DescribeStackSetOperationInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackSetOperationInput = exports.DescribeStackSetOperationInput || (exports.DescribeStackSetOperationInput = {}));
var StackSetOperation;
(function (StackSetOperation) {
    StackSetOperation.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetOperation = exports.StackSetOperation || (exports.StackSetOperation = {}));
var DescribeStackSetOperationOutput;
(function (DescribeStackSetOperationOutput) {
    DescribeStackSetOperationOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeStackSetOperationOutput = exports.DescribeStackSetOperationOutput || (exports.DescribeStackSetOperationOutput = {}));
var OperationNotFoundException;
(function (OperationNotFoundException) {
    OperationNotFoundException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OperationNotFoundException = exports.OperationNotFoundException || (exports.OperationNotFoundException = {}));
var DescribeTypeInput;
(function (DescribeTypeInput) {
    DescribeTypeInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeTypeInput = exports.DescribeTypeInput || (exports.DescribeTypeInput = {}));
var RequiredActivatedType;
(function (RequiredActivatedType) {
    RequiredActivatedType.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RequiredActivatedType = exports.RequiredActivatedType || (exports.RequiredActivatedType = {}));
var TypeTestsStatus;
(function (TypeTestsStatus) {
    TypeTestsStatus["FAILED"] = "FAILED";
    TypeTestsStatus["IN_PROGRESS"] = "IN_PROGRESS";
    TypeTestsStatus["NOT_TESTED"] = "NOT_TESTED";
    TypeTestsStatus["PASSED"] = "PASSED";
})(TypeTestsStatus = exports.TypeTestsStatus || (exports.TypeTestsStatus = {}));
var DescribeTypeOutput;
(function (DescribeTypeOutput) {
    DescribeTypeOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeTypeOutput = exports.DescribeTypeOutput || (exports.DescribeTypeOutput = {}));
var DescribeTypeRegistrationInput;
(function (DescribeTypeRegistrationInput) {
    DescribeTypeRegistrationInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeTypeRegistrationInput = exports.DescribeTypeRegistrationInput || (exports.DescribeTypeRegistrationInput = {}));
var DescribeTypeRegistrationOutput;
(function (DescribeTypeRegistrationOutput) {
    DescribeTypeRegistrationOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DescribeTypeRegistrationOutput = exports.DescribeTypeRegistrationOutput || (exports.DescribeTypeRegistrationOutput = {}));
var DetectStackDriftInput;
(function (DetectStackDriftInput) {
    DetectStackDriftInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectStackDriftInput = exports.DetectStackDriftInput || (exports.DetectStackDriftInput = {}));
var DetectStackDriftOutput;
(function (DetectStackDriftOutput) {
    DetectStackDriftOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectStackDriftOutput = exports.DetectStackDriftOutput || (exports.DetectStackDriftOutput = {}));
var DetectStackResourceDriftInput;
(function (DetectStackResourceDriftInput) {
    DetectStackResourceDriftInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectStackResourceDriftInput = exports.DetectStackResourceDriftInput || (exports.DetectStackResourceDriftInput = {}));
var DetectStackResourceDriftOutput;
(function (DetectStackResourceDriftOutput) {
    DetectStackResourceDriftOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectStackResourceDriftOutput = exports.DetectStackResourceDriftOutput || (exports.DetectStackResourceDriftOutput = {}));
var DetectStackSetDriftInput;
(function (DetectStackSetDriftInput) {
    DetectStackSetDriftInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectStackSetDriftInput = exports.DetectStackSetDriftInput || (exports.DetectStackSetDriftInput = {}));
var DetectStackSetDriftOutput;
(function (DetectStackSetDriftOutput) {
    DetectStackSetDriftOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(DetectStackSetDriftOutput = exports.DetectStackSetDriftOutput || (exports.DetectStackSetDriftOutput = {}));
var EstimateTemplateCostInput;
(function (EstimateTemplateCostInput) {
    EstimateTemplateCostInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EstimateTemplateCostInput = exports.EstimateTemplateCostInput || (exports.EstimateTemplateCostInput = {}));
var EstimateTemplateCostOutput;
(function (EstimateTemplateCostOutput) {
    EstimateTemplateCostOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(EstimateTemplateCostOutput = exports.EstimateTemplateCostOutput || (exports.EstimateTemplateCostOutput = {}));
var ExecuteChangeSetInput;
(function (ExecuteChangeSetInput) {
    ExecuteChangeSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ExecuteChangeSetInput = exports.ExecuteChangeSetInput || (exports.ExecuteChangeSetInput = {}));
var ExecuteChangeSetOutput;
(function (ExecuteChangeSetOutput) {
    ExecuteChangeSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ExecuteChangeSetOutput = exports.ExecuteChangeSetOutput || (exports.ExecuteChangeSetOutput = {}));
var GetStackPolicyInput;
(function (GetStackPolicyInput) {
    GetStackPolicyInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStackPolicyInput = exports.GetStackPolicyInput || (exports.GetStackPolicyInput = {}));
var GetStackPolicyOutput;
(function (GetStackPolicyOutput) {
    GetStackPolicyOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetStackPolicyOutput = exports.GetStackPolicyOutput || (exports.GetStackPolicyOutput = {}));
var TemplateStage;
(function (TemplateStage) {
    TemplateStage["Original"] = "Original";
    TemplateStage["Processed"] = "Processed";
})(TemplateStage = exports.TemplateStage || (exports.TemplateStage = {}));
var GetTemplateInput;
(function (GetTemplateInput) {
    GetTemplateInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTemplateInput = exports.GetTemplateInput || (exports.GetTemplateInput = {}));
var GetTemplateOutput;
(function (GetTemplateOutput) {
    GetTemplateOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTemplateOutput = exports.GetTemplateOutput || (exports.GetTemplateOutput = {}));
var GetTemplateSummaryInput;
(function (GetTemplateSummaryInput) {
    GetTemplateSummaryInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTemplateSummaryInput = exports.GetTemplateSummaryInput || (exports.GetTemplateSummaryInput = {}));
var ParameterConstraints;
(function (ParameterConstraints) {
    ParameterConstraints.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ParameterConstraints = exports.ParameterConstraints || (exports.ParameterConstraints = {}));
var ParameterDeclaration;
(function (ParameterDeclaration) {
    ParameterDeclaration.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ParameterDeclaration = exports.ParameterDeclaration || (exports.ParameterDeclaration = {}));
var ResourceIdentifierSummary;
(function (ResourceIdentifierSummary) {
    ResourceIdentifierSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ResourceIdentifierSummary = exports.ResourceIdentifierSummary || (exports.ResourceIdentifierSummary = {}));
var GetTemplateSummaryOutput;
(function (GetTemplateSummaryOutput) {
    GetTemplateSummaryOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(GetTemplateSummaryOutput = exports.GetTemplateSummaryOutput || (exports.GetTemplateSummaryOutput = {}));
var ImportStacksToStackSetInput;
(function (ImportStacksToStackSetInput) {
    ImportStacksToStackSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ImportStacksToStackSetInput = exports.ImportStacksToStackSetInput || (exports.ImportStacksToStackSetInput = {}));
var ImportStacksToStackSetOutput;
(function (ImportStacksToStackSetOutput) {
    ImportStacksToStackSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ImportStacksToStackSetOutput = exports.ImportStacksToStackSetOutput || (exports.ImportStacksToStackSetOutput = {}));
var StackNotFoundException;
(function (StackNotFoundException) {
    StackNotFoundException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackNotFoundException = exports.StackNotFoundException || (exports.StackNotFoundException = {}));
var ListChangeSetsInput;
(function (ListChangeSetsInput) {
    ListChangeSetsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListChangeSetsInput = exports.ListChangeSetsInput || (exports.ListChangeSetsInput = {}));
var ListChangeSetsOutput;
(function (ListChangeSetsOutput) {
    ListChangeSetsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListChangeSetsOutput = exports.ListChangeSetsOutput || (exports.ListChangeSetsOutput = {}));
var ListExportsInput;
(function (ListExportsInput) {
    ListExportsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListExportsInput = exports.ListExportsInput || (exports.ListExportsInput = {}));
var Export;
(function (Export) {
    Export.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(Export = exports.Export || (exports.Export = {}));
var ListExportsOutput;
(function (ListExportsOutput) {
    ListExportsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListExportsOutput = exports.ListExportsOutput || (exports.ListExportsOutput = {}));
var ListImportsInput;
(function (ListImportsInput) {
    ListImportsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListImportsInput = exports.ListImportsInput || (exports.ListImportsInput = {}));
var ListImportsOutput;
(function (ListImportsOutput) {
    ListImportsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListImportsOutput = exports.ListImportsOutput || (exports.ListImportsOutput = {}));
var StackInstanceFilter;
(function (StackInstanceFilter) {
    StackInstanceFilter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackInstanceFilter = exports.StackInstanceFilter || (exports.StackInstanceFilter = {}));
var ListStackInstancesInput;
(function (ListStackInstancesInput) {
    ListStackInstancesInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackInstancesInput = exports.ListStackInstancesInput || (exports.ListStackInstancesInput = {}));
var StackInstanceSummary;
(function (StackInstanceSummary) {
    StackInstanceSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackInstanceSummary = exports.StackInstanceSummary || (exports.StackInstanceSummary = {}));
var ListStackInstancesOutput;
(function (ListStackInstancesOutput) {
    ListStackInstancesOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackInstancesOutput = exports.ListStackInstancesOutput || (exports.ListStackInstancesOutput = {}));
var ListStackResourcesInput;
(function (ListStackResourcesInput) {
    ListStackResourcesInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackResourcesInput = exports.ListStackResourcesInput || (exports.ListStackResourcesInput = {}));
var StackResourceDriftInformationSummary;
(function (StackResourceDriftInformationSummary) {
    StackResourceDriftInformationSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackResourceDriftInformationSummary = exports.StackResourceDriftInformationSummary || (exports.StackResourceDriftInformationSummary = {}));
var StackResourceSummary;
(function (StackResourceSummary) {
    StackResourceSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackResourceSummary = exports.StackResourceSummary || (exports.StackResourceSummary = {}));
var ListStackResourcesOutput;
(function (ListStackResourcesOutput) {
    ListStackResourcesOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackResourcesOutput = exports.ListStackResourcesOutput || (exports.ListStackResourcesOutput = {}));
var ListStacksInput;
(function (ListStacksInput) {
    ListStacksInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStacksInput = exports.ListStacksInput || (exports.ListStacksInput = {}));
var StackDriftInformationSummary;
(function (StackDriftInformationSummary) {
    StackDriftInformationSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackDriftInformationSummary = exports.StackDriftInformationSummary || (exports.StackDriftInformationSummary = {}));
var StackSummary;
(function (StackSummary) {
    StackSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSummary = exports.StackSummary || (exports.StackSummary = {}));
var ListStacksOutput;
(function (ListStacksOutput) {
    ListStacksOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStacksOutput = exports.ListStacksOutput || (exports.ListStacksOutput = {}));
var ListStackSetOperationResultsInput;
(function (ListStackSetOperationResultsInput) {
    ListStackSetOperationResultsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackSetOperationResultsInput = exports.ListStackSetOperationResultsInput || (exports.ListStackSetOperationResultsInput = {}));
var StackSetOperationResultSummary;
(function (StackSetOperationResultSummary) {
    StackSetOperationResultSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetOperationResultSummary = exports.StackSetOperationResultSummary || (exports.StackSetOperationResultSummary = {}));
var ListStackSetOperationResultsOutput;
(function (ListStackSetOperationResultsOutput) {
    ListStackSetOperationResultsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackSetOperationResultsOutput = exports.ListStackSetOperationResultsOutput || (exports.ListStackSetOperationResultsOutput = {}));
var ListStackSetOperationsInput;
(function (ListStackSetOperationsInput) {
    ListStackSetOperationsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackSetOperationsInput = exports.ListStackSetOperationsInput || (exports.ListStackSetOperationsInput = {}));
var StackSetOperationSummary;
(function (StackSetOperationSummary) {
    StackSetOperationSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetOperationSummary = exports.StackSetOperationSummary || (exports.StackSetOperationSummary = {}));
var ListStackSetOperationsOutput;
(function (ListStackSetOperationsOutput) {
    ListStackSetOperationsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackSetOperationsOutput = exports.ListStackSetOperationsOutput || (exports.ListStackSetOperationsOutput = {}));
var ListStackSetsInput;
(function (ListStackSetsInput) {
    ListStackSetsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackSetsInput = exports.ListStackSetsInput || (exports.ListStackSetsInput = {}));
var StackSetSummary;
(function (StackSetSummary) {
    StackSetSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StackSetSummary = exports.StackSetSummary || (exports.StackSetSummary = {}));
var ListStackSetsOutput;
(function (ListStackSetsOutput) {
    ListStackSetsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListStackSetsOutput = exports.ListStackSetsOutput || (exports.ListStackSetsOutput = {}));
var ListTypeRegistrationsInput;
(function (ListTypeRegistrationsInput) {
    ListTypeRegistrationsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTypeRegistrationsInput = exports.ListTypeRegistrationsInput || (exports.ListTypeRegistrationsInput = {}));
var ListTypeRegistrationsOutput;
(function (ListTypeRegistrationsOutput) {
    ListTypeRegistrationsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTypeRegistrationsOutput = exports.ListTypeRegistrationsOutput || (exports.ListTypeRegistrationsOutput = {}));
var TypeFilters;
(function (TypeFilters) {
    TypeFilters.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TypeFilters = exports.TypeFilters || (exports.TypeFilters = {}));
var ListTypesInput;
(function (ListTypesInput) {
    ListTypesInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTypesInput = exports.ListTypesInput || (exports.ListTypesInput = {}));
var TypeSummary;
(function (TypeSummary) {
    TypeSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TypeSummary = exports.TypeSummary || (exports.TypeSummary = {}));
var ListTypesOutput;
(function (ListTypesOutput) {
    ListTypesOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTypesOutput = exports.ListTypesOutput || (exports.ListTypesOutput = {}));
var ListTypeVersionsInput;
(function (ListTypeVersionsInput) {
    ListTypeVersionsInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTypeVersionsInput = exports.ListTypeVersionsInput || (exports.ListTypeVersionsInput = {}));
var TypeVersionSummary;
(function (TypeVersionSummary) {
    TypeVersionSummary.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TypeVersionSummary = exports.TypeVersionSummary || (exports.TypeVersionSummary = {}));
var ListTypeVersionsOutput;
(function (ListTypeVersionsOutput) {
    ListTypeVersionsOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ListTypeVersionsOutput = exports.ListTypeVersionsOutput || (exports.ListTypeVersionsOutput = {}));
var PublishTypeInput;
(function (PublishTypeInput) {
    PublishTypeInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublishTypeInput = exports.PublishTypeInput || (exports.PublishTypeInput = {}));
var PublishTypeOutput;
(function (PublishTypeOutput) {
    PublishTypeOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(PublishTypeOutput = exports.PublishTypeOutput || (exports.PublishTypeOutput = {}));
var InvalidStateTransitionException;
(function (InvalidStateTransitionException) {
    InvalidStateTransitionException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(InvalidStateTransitionException = exports.InvalidStateTransitionException || (exports.InvalidStateTransitionException = {}));
var OperationStatusCheckFailedException;
(function (OperationStatusCheckFailedException) {
    OperationStatusCheckFailedException.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(OperationStatusCheckFailedException = exports.OperationStatusCheckFailedException || (exports.OperationStatusCheckFailedException = {}));
var OperationStatus;
(function (OperationStatus) {
    OperationStatus["FAILED"] = "FAILED";
    OperationStatus["IN_PROGRESS"] = "IN_PROGRESS";
    OperationStatus["PENDING"] = "PENDING";
    OperationStatus["SUCCESS"] = "SUCCESS";
})(OperationStatus = exports.OperationStatus || (exports.OperationStatus = {}));
var HandlerErrorCode;
(function (HandlerErrorCode) {
    HandlerErrorCode["AccessDenied"] = "AccessDenied";
    HandlerErrorCode["AlreadyExists"] = "AlreadyExists";
    HandlerErrorCode["GeneralServiceException"] = "GeneralServiceException";
    HandlerErrorCode["InternalFailure"] = "InternalFailure";
    HandlerErrorCode["InvalidCredentials"] = "InvalidCredentials";
    HandlerErrorCode["InvalidRequest"] = "InvalidRequest";
    HandlerErrorCode["InvalidTypeConfiguration"] = "InvalidTypeConfiguration";
    HandlerErrorCode["NetworkFailure"] = "NetworkFailure";
    HandlerErrorCode["NotFound"] = "NotFound";
    HandlerErrorCode["NotUpdatable"] = "NotUpdatable";
    HandlerErrorCode["ResourceConflict"] = "ResourceConflict";
    HandlerErrorCode["ServiceInternalError"] = "ServiceInternalError";
    HandlerErrorCode["ServiceLimitExceeded"] = "ServiceLimitExceeded";
    HandlerErrorCode["ServiceTimeout"] = "NotStabilized";
    HandlerErrorCode["Throttling"] = "Throttling";
})(HandlerErrorCode = exports.HandlerErrorCode || (exports.HandlerErrorCode = {}));
var RecordHandlerProgressInput;
(function (RecordHandlerProgressInput) {
    RecordHandlerProgressInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RecordHandlerProgressInput = exports.RecordHandlerProgressInput || (exports.RecordHandlerProgressInput = {}));
var RecordHandlerProgressOutput;
(function (RecordHandlerProgressOutput) {
    RecordHandlerProgressOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RecordHandlerProgressOutput = exports.RecordHandlerProgressOutput || (exports.RecordHandlerProgressOutput = {}));
var RegisterPublisherInput;
(function (RegisterPublisherInput) {
    RegisterPublisherInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RegisterPublisherInput = exports.RegisterPublisherInput || (exports.RegisterPublisherInput = {}));
var RegisterPublisherOutput;
(function (RegisterPublisherOutput) {
    RegisterPublisherOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RegisterPublisherOutput = exports.RegisterPublisherOutput || (exports.RegisterPublisherOutput = {}));
var RegisterTypeInput;
(function (RegisterTypeInput) {
    RegisterTypeInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RegisterTypeInput = exports.RegisterTypeInput || (exports.RegisterTypeInput = {}));
var RegisterTypeOutput;
(function (RegisterTypeOutput) {
    RegisterTypeOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RegisterTypeOutput = exports.RegisterTypeOutput || (exports.RegisterTypeOutput = {}));
var RollbackStackInput;
(function (RollbackStackInput) {
    RollbackStackInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RollbackStackInput = exports.RollbackStackInput || (exports.RollbackStackInput = {}));
var RollbackStackOutput;
(function (RollbackStackOutput) {
    RollbackStackOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(RollbackStackOutput = exports.RollbackStackOutput || (exports.RollbackStackOutput = {}));
var SetStackPolicyInput;
(function (SetStackPolicyInput) {
    SetStackPolicyInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SetStackPolicyInput = exports.SetStackPolicyInput || (exports.SetStackPolicyInput = {}));
var SetTypeConfigurationInput;
(function (SetTypeConfigurationInput) {
    SetTypeConfigurationInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SetTypeConfigurationInput = exports.SetTypeConfigurationInput || (exports.SetTypeConfigurationInput = {}));
var SetTypeConfigurationOutput;
(function (SetTypeConfigurationOutput) {
    SetTypeConfigurationOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SetTypeConfigurationOutput = exports.SetTypeConfigurationOutput || (exports.SetTypeConfigurationOutput = {}));
var SetTypeDefaultVersionInput;
(function (SetTypeDefaultVersionInput) {
    SetTypeDefaultVersionInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SetTypeDefaultVersionInput = exports.SetTypeDefaultVersionInput || (exports.SetTypeDefaultVersionInput = {}));
var SetTypeDefaultVersionOutput;
(function (SetTypeDefaultVersionOutput) {
    SetTypeDefaultVersionOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SetTypeDefaultVersionOutput = exports.SetTypeDefaultVersionOutput || (exports.SetTypeDefaultVersionOutput = {}));
var ResourceSignalStatus;
(function (ResourceSignalStatus) {
    ResourceSignalStatus["FAILURE"] = "FAILURE";
    ResourceSignalStatus["SUCCESS"] = "SUCCESS";
})(ResourceSignalStatus = exports.ResourceSignalStatus || (exports.ResourceSignalStatus = {}));
var SignalResourceInput;
(function (SignalResourceInput) {
    SignalResourceInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(SignalResourceInput = exports.SignalResourceInput || (exports.SignalResourceInput = {}));
var StopStackSetOperationInput;
(function (StopStackSetOperationInput) {
    StopStackSetOperationInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopStackSetOperationInput = exports.StopStackSetOperationInput || (exports.StopStackSetOperationInput = {}));
var StopStackSetOperationOutput;
(function (StopStackSetOperationOutput) {
    StopStackSetOperationOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(StopStackSetOperationOutput = exports.StopStackSetOperationOutput || (exports.StopStackSetOperationOutput = {}));
var TestTypeInput;
(function (TestTypeInput) {
    TestTypeInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TestTypeInput = exports.TestTypeInput || (exports.TestTypeInput = {}));
var TestTypeOutput;
(function (TestTypeOutput) {
    TestTypeOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TestTypeOutput = exports.TestTypeOutput || (exports.TestTypeOutput = {}));
var UpdateStackInput;
(function (UpdateStackInput) {
    UpdateStackInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStackInput = exports.UpdateStackInput || (exports.UpdateStackInput = {}));
var UpdateStackOutput;
(function (UpdateStackOutput) {
    UpdateStackOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStackOutput = exports.UpdateStackOutput || (exports.UpdateStackOutput = {}));
var UpdateStackInstancesInput;
(function (UpdateStackInstancesInput) {
    UpdateStackInstancesInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStackInstancesInput = exports.UpdateStackInstancesInput || (exports.UpdateStackInstancesInput = {}));
var UpdateStackInstancesOutput;
(function (UpdateStackInstancesOutput) {
    UpdateStackInstancesOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStackInstancesOutput = exports.UpdateStackInstancesOutput || (exports.UpdateStackInstancesOutput = {}));
var UpdateStackSetInput;
(function (UpdateStackSetInput) {
    UpdateStackSetInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStackSetInput = exports.UpdateStackSetInput || (exports.UpdateStackSetInput = {}));
var UpdateStackSetOutput;
(function (UpdateStackSetOutput) {
    UpdateStackSetOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateStackSetOutput = exports.UpdateStackSetOutput || (exports.UpdateStackSetOutput = {}));
var UpdateTerminationProtectionInput;
(function (UpdateTerminationProtectionInput) {
    UpdateTerminationProtectionInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTerminationProtectionInput = exports.UpdateTerminationProtectionInput || (exports.UpdateTerminationProtectionInput = {}));
var UpdateTerminationProtectionOutput;
(function (UpdateTerminationProtectionOutput) {
    UpdateTerminationProtectionOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(UpdateTerminationProtectionOutput = exports.UpdateTerminationProtectionOutput || (exports.UpdateTerminationProtectionOutput = {}));
var ValidateTemplateInput;
(function (ValidateTemplateInput) {
    ValidateTemplateInput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ValidateTemplateInput = exports.ValidateTemplateInput || (exports.ValidateTemplateInput = {}));
var TemplateParameter;
(function (TemplateParameter) {
    TemplateParameter.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(TemplateParameter = exports.TemplateParameter || (exports.TemplateParameter = {}));
var ValidateTemplateOutput;
(function (ValidateTemplateOutput) {
    ValidateTemplateOutput.filterSensitiveLog = (obj) => ({
        ...obj,
    });
})(ValidateTemplateOutput = exports.ValidateTemplateOutput || (exports.ValidateTemplateOutput = {}));
